internal/controllers/v1beta2/buildpodmonitor_handlers.go

```diff
@@ -508,11 +508,45 @@ Build %s
 			allContainerLogs = logs
 		}
 
+		published := false
+		if helpers.ContainsString(
+			lagooncrd.BuildCompletedCancelledFailedStatus,
+			lagoonBuild.Labels["lagoon.sh/buildStatus"],
+		) {
+			// if the build has completed, failed, or been cancelled
+			published = true
+			// check if this build has already published data or not
+			if value, ok := lagoonBuild.Labels["build.lagoon.sh/published"]; ok {
+				published, _ = strconv.ParseBool(value)
+			}
+		}
+		// if it hasn't published after completion, do that now
+		if !published {
```

Comment:
okay - I 'm a little confused about the logic in this section (524-549)
This conditional will only ever trigger when `published==false`, since the condition is basically `if NOT(published)` (so published = false).
But, you're only ever setting `published` to false in this - at no point are you setting it to true.
So, unless there's some weird closure stuff happening, it seems that regardless of the outcome, `published` looks to be "false" at the end of this.
At the very least, it speaks to these assignments on 528, 532, and 539 being redundant.

Reply:
Turns out, this should have been removed after the other changes made.